solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -45,10 +49,501 @@ describe("Bridge - Moving funds", () => {
 
   before(async () => {
     // eslint-disable-next-line @typescript-eslint/no-extra-semi
-    ;({ treasury, bank, relay, walletRegistry, Bridge, bridge } =
+    ;({ thirdParty, treasury, bank, relay, walletRegistry, Bridge, bridge } =
       await waffle.loadFixture(fixture))
   })
 
+  describe("submitMovingFundsCommitment", () => {
+    const walletDraft = {
+      ecdsaWalletID: ecdsaWalletTestData.walletID,
+      mainUtxoHash: ethers.constants.HashZero,
+      pendingRedemptionsValue: 0,
+      createdAt: 0,
+      movingFundsRequestedAt: 0,
+      state: walletState.Unknown,
+      movingFundsTargetWalletsCommitmentHash: ethers.constants.HashZero,
+    }
+
+    context("when source wallet is in the MovingFunds state", () => {
+      before(async () => {
+        await createSnapshot()
+
+        await bridge.setWallet(ecdsaWalletTestData.pubKeyHash160, {
+          ...walletDraft,
+          state: walletState.MovingFunds,
+        })
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      context("when source wallet has no pending redemptions", () => {
+        // The wallet created using the `walletDraft` has no pending redemptions
+        // by default. No need to do anything here.
+
+        context("when the commitment was not submitted yet", () => {
+          // The wallet created using the `walletDraft` has no commitment
+          // submitted by default. No need to do anything here.
+
+          context("when the caller is a member of the source wallet", () => {
+            const walletMembersIDs = [1, 2, 3, 4, 5]
+            const walletMemberIndex = 2
+
+            let caller
```

Comment:
I would give it a type, so that it's not of type `any`:
```
let caller: SignerWithAddress
```

Reply:
See https://github.com/keep-network/tbtc-v2/pull/208/commits/26678877b05f029adf65a6c168f407e25b78bd3b